SpiffWorkflow/SpiffWorkflow/bpmn/workflow.py

```diff
@@ -127,6 +128,10 @@ def catch(self, event_definition, correlations=None):
         should call the event_definition's reset method before executing to
         clear out a stale message.
 
+        We might be catching an event that was thrown from some other part of
+        our own workflow, and it needs to continue out, but if it originated
+        externally, we should not pass it on.
+
```

Comment:
not certain how to resolve this then -- could use some help identifying a failing test around this code.

Reply:
I think I misunderstood what you were trying to accomplish.  There could be a call activity or subprocess waiting for a message generated externally and those definitely need to be passed on; however, it does make sense to check whether the message is external when deciding whether to add it back to external messages.  I don't think `event_definition.catch` needs to know about this at all, but it does make sense to prevent external messages being added back into the external message queue.